ldapauthenticator/ldapauthenticator.py

```diff
@@ -463,17 +462,12 @@ async def authenticate(self, handler, data):
             username, resolved_dn = self.resolve_username(username)
             if not username:
                 return None
-            if str(self.lookup_dn_user_dn_attribute).upper() == "CN":
-                # Only escape commas if the lookup attribute is CN
-                username = re.subn(r"([^\\]),", r"\1\,", username)[0]
```

Comment:
The later escape_rdn is a superset of this comma escape. So probably ok. 

However the if ... logic is now missing. Is it important to only do this escape at some times?

Reply:
The if logic didn't make sense to me, the username should reasonably always be escaped before being passed to the LDAP server.

Hmmm, i guess this could also have impacted the username chosen if use_lookup_dn_username is true as well though...

Are you a user of that?

Reply:
I think for my use case the if condition would always be true as my company uses Active Directory.

Reply:
Hmmm, the issue I'm seeing now, is that `username` is assigned with the escaping, and the username variable will, conditionally on `use_lookup_dn_username` end up being used to become the JupyterHub username, which otherwise can become whatever the user initially wrote in the login prompt.

This is because of this line of code, where `resolved_username` seen below was previously named `username`.

https://github.com/jupyterhub/ldapauthenticator/blob/cea8616e2b65d8ef75c533900b3029663586158b/ldapauthenticator/ldapauthenticator.py#L642-L647

Reply:
I updated the PR description with this:

## EDIT: Breaking change

This PR introduced a breaking change in a edge case scenario if both...

- the following LDAPAuthenticator configuration was used: `lookup_dn = True`, `lookup_dn_user_dn_attribute = "cn"`, `use_lookup_dn_username = True` (previous default value)
- one or more users, that had already signed in at any point in time before, had a comma in the cn attribute's value, as they would if for example the cn attribute values looks like "lastname, firstname"

Then, these users' resulting JupyterHub usernames would be changed from looking like `"lastname\\, firstname"` to `"lastname, firstname"`.